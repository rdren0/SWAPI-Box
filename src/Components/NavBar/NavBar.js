import React from 'react';

const Header = ({filterByType}) => {
  const filterType = (e) =>{
    filterByType(e.target.innerText)

  }
  return(
    <div>
      <button id="people" onClick={filterType}> People </button>
      <button id="planets" onClick={filterType}> Planets </button>
      <button id="vehicles" onClick={filterType}> Vehicles </button>
    </div>)
}

export default Header


 
  