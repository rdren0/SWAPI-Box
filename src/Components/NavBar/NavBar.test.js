import React, { Component } from 'react';
import NavBar from './NavBar.js';
import { shallow } from 'enzyme';



describe(NavBar, () => {
  let wrapper;
  let mockFilterType;
  let mockPeopleEvent;
  let mockPlanetEvent;
  let mockVehicleEvent;


 beforeEach(() => {
    mockFilterType = jest.fn()
    mockPeopleEvent = {target:{innerText:'people'}}
    mockPlanetEvent = {target:{innerText:'planets'}}
    mockVehicleEvent = {target:{innerText:'vehicles'}}


    wrapper = shallow(<NavBar filterByType={mockFilterType}/>)
  })

  it('should match snapshot with all correct data passed through', () => {
    const wrapper = shallow(<NavBar/>)
    expect(wrapper).toMatchSnapshot();
  })
  it.skip('should call filterType with correct params when clicked', () => {
    wrapper.find('#people').simulate('click');
    console.log(mockEvent)
    expect(mockFilterType).toHaveBeenCalledWith(mockPeopleEvent)
  });
  it.skip('should call filterType with correct params when clicked', () => {
    wrapper.find('#planets').simulate('click');
    expect(mockFilterType).toHaveBeenCalledWith(mockPlanetEvent)
  });
  it.skip('should call filterType with correct params when clicked', () => {
    wrapper.find('#vehicles').simulate('click');
    expect(mockFilterType).toHaveBeenCalledWith(mockVehicleEvent)
  });

})

