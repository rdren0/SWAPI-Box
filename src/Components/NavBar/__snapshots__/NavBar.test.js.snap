// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot with all correct data passed through 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          id="people"
          onClick={[Function]}
        >
           People 
        </button>,
        <button
          id="planets"
          onClick={[Function]}
        >
           Planets 
        </button>,
        <button
          id="vehicles"
          onClick={[Function]}
        >
           Vehicles 
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " People ",
          "id": "people",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": " People ",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Planets ",
          "id": "planets",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": " Planets ",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Vehicles ",
          "id": "vehicles",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": " Vehicles ",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            id="people"
            onClick={[Function]}
          >
             People 
          </button>,
          <button
            id="planets"
            onClick={[Function]}
          >
             Planets 
          </button>,
          <button
            id="vehicles"
            onClick={[Function]}
          >
             Vehicles 
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " People ",
            "id": "people",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": " People ",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Planets ",
            "id": "planets",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": " Planets ",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Vehicles ",
            "id": "vehicles",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": " Vehicles ",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
